django/control/api/update_batch.py

```diff
@@ -0,0 +1,155 @@
+        batch_counter_redis.sadd(f"update-batch:{batch_id}:resources", *resource_ids)
+
+        try:
+            # Create kafka topics for batch
+            new_topic_names = [f"batch.{batch_id}", f"extract.{batch_id}", f"transform.{batch_id}", f"load.{batch_id}"]
+            create_kafka_topics(new_topic_names)
+            # Send event to the extractor
+            send_batch_events(batch_id, resource_ids)
+        except (KafkaException, ValueError) as err:
+            logger.exception(err)
+            # Clean the batch
+            TopicleanerHandler().delete_batch(batch_id)
+            return Response(
+                {"id": batch_id, "error": "error while producing extract events"},
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            )
+
+        # Update Airflow variable to create new DAG
+        # Give batch id (for the mapping) and freq of refresh in dict
+        airflow_client = AirflowClient()
+        try:
+            get_variable_resp = airflow_client.get(f"variables/{settings.AIRFLOW_UPDATE_VARIABLE}")
+            cur_update_variable_value = get_variable_resp.json()["value"]
+            new_update_variable_value = {batch_id: data["schedule_interval"], **cur_update_variable_value}
+
+            airflow_client.post("variables", json=new_update_variable_value)
+        except (HTTPError, AirflowQueryStatusCodeException):
+            return Response(
+                {"id": batch_id, "error": "error while updating Airflow variable"},
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            )
+
+        return Response({"id": batch_id, "timestamp": batch_timestamp}, status=status.HTTP_200_OK)
+
+    def retrieve(self, request, pk=None):
+        """Route used to create an "update" batch"""
+        batch_timestamp = datetime.now().strftime("%Y-%m-%dT%H:%M:%S")
+
+        batch_counter_redis = redis.Redis(
+            host=settings.REDIS_COUNTER_HOST, port=settings.REDIS_COUNTER_PORT, db=settings.REDIS_COUNTER_DB
+        )
+        resource_ids = batch_counter_redis.smembers(f"update-batch:{pk}:resources")
+
+        try:
+            # Create kafka topics for batch
+            # TODO do we want to keep the batch between 2 updates?
+            new_topic_names = [f"batch.{pk}", f"extract.{pk}", f"transform.{pk}", f"load.{pk}"]
+            create_kafka_topics(new_topic_names)
+            # Send event to the extractor
+            send_batch_events(pk, resource_ids)
+        except (KafkaException, ValueError) as err:
+            logger.exception(err)
+            # Clean the batch
+            TopicleanerHandler().delete_batch(pk)
+            return Response(
+                {"id": pk, "error": "error while producing extract events"},
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            )
+
+        return Response({"id": pk, "timestamp": batch_timestamp}, status=status.HTTP_200_OK)
+
+    def destroy(self, request, pk=None):
```

Comment:
remind me why we don't use `TopicleanerHandler().delete_batch(batch_id)` here ?

Reply:
Legacy, fixed that